graph/patterns/default-properties.md

```diff
@@ -0,0 +1,201 @@
+# Default properties
+
+Microsoft Graph API Design Pattern
+
+*The default properties pattern allows API producers to omit specific properties from the response unless they are explicitly requested using `$select`. The pattern also uses an instance annotation to inform callers that other properties are also available. The same annotation is also use to encourage callers to use `$select`.*
+
+## Problem
+
+API producers want to control the set of properties that their entities return by default, when the incoming request does not specify a `$select`. This can be desirable when an entity type has many properties or an API producer needs to add properties that are computationally expensive to return by default.
+
+## Solution
+
+For incoming requests targeting an entity type where the caller does not specify a `$select`` clause, API producers **may** return a subset of the entity type's properties, omitting computationally expensive properties.
+
+To get the non-default properties of an entity type, callers must explicitly request them using `$select`.
```

Comment:
I want to capture my concern here that our current solution is not discoverable at the API design time, and therefore downstream tools like documentation, GE and SDKs can only implement dynamic runtime support. This basically excludes documentation from the picture. It also makes the API behavior less predictable upfront and complicates the developer experience. I suggest adding an additional annotation to specify a default set of properties at the design time.

Reply:
To play devils advocate for a moment -- why are we telling developers which properties are returned by default and which they have to $select.  Shouldn't the guidance be to always select exactly the set of properties that you need?  Do we expect them to write code that normally omits $select, but conditionally adds it if they need a non-default property?

Reply:
Triage: will leave as is and add a new work item for additional annotations for default properties.

Reply:
Filing a work item for this